### ISSUE TITLE
[Documentation] Explain how to add patches 

### ISSUE THREAD

**Issue by ShalokShalom** [habitat-sh/habitat#2583]
See also: https://github.com/habitat-sh/habitat/issues/2561#issuecomment-308230169

**Comment by eeyun** [comment]
Looks like my original comment didn't get submitted appropriately. Whoops!

We should likely expose that habitat supports build time patching like other software build systems and perhaps link to the core plans that are utilizing that kind of patching behavior.

I'd heartily suggest avoiding diving too deep into this because patching software will be completely dependent upon the type of software being built and those behaviors are not specific to habitat.

**Comment by ShalokShalom** [comment]
Then i am a little bit curious, since i thought that i can simply reuse the patches of specific distributions in an easy way. So, this means easy patching like in PKGBUILDs will never be the case?

Regarding to the closed issue. How that? Isnt patching something very essential in packaging?

**Comment by reset** [comment]
@ShalokShalom sure, you can re-use patches from anyone including other distributions. What @eeyun is suggesting is that applying patches to software isn't a direct concern of Habitat itself but instead of the [patch tool](https://en.wikipedia.org/wiki/Patch_(Unix)) itself.

I don't mind if this makes it into some sort of "deep continued learning section" of our docs but we shouldn't try to recreate man pages for Patch itself ;)

**Comment by ShalokShalom** [comment]
I really love to see a dedicated patch function in Habitat, which got refused in the past :P

Thanks, i will close this.

**Comment by ShalokShalom** [comment]
This is a great example: 

https://github.com/habitat-sh/core-plans/tree/master/bash

**Comment by robbkidd** [comment]
I reckon the docs could demonstrate at least the two patching patterns—[patching with patches versioned with the plan](https://github.com/habitat-sh/core-plans/tree/cc5e89933245c04aec345261cbf83f509b0769e8/glibc) and [patching with patches retrieved from elsewhere at build time](https://github.com/habitat-sh/core-plans/tree/cc5e89933245c04aec345261cbf83f509b0769e8/bash)—shown in the plans linked to by @eeyun over on #2561 (😵  hard to follow all the issues opened about this).

**Comment by ShalokShalom** [comment]
The proposal here is generic and lend by the bash version.

**Comment by robbkidd** [comment]
Indeed. That is probably roughly what would get documented for an example of retrieving and applying patches at build time.

**Comment by ShalokShalom** [comment]
So far as I know, got Habitat proper patch support, a few months after this issue was raised.
